website/ref/language.md

```diff
@@ -106,16 +106,75 @@ error.
 
 The three syntaxes above all evaluate to strings, and they are interchangeable.
 For instance, `xyz`, `'xyz'` and `"xyz"` are different syntaxes for the same
-string, and they are always equivalent.
+string, and they are always equivalent with the exception of
+**escape sequences** as documented above.
 
-Elvish does have a dedicated `float64` number type, but it does not have a
-literal syntax; instead it needs to be constructed using a function call, as in
-`put (float64 42)`; the argument `42` is a string. For convenience, all Elvish
-functions that expect number arguments also accept strings that can be parsed as
-a number, so explicit number constructors are not needed often.
+# Numeric data type
 
+Elvish has a numeric data type, `float64`, created using this syntax:
+`(float64 numeric-literal)`. For convenience, all Elvish functions that
+expect a numeric argument will accept a string that can be parsed as a
+number. They will automatically convert the string to a `float64`. Which
+means explicit `float64` constructors are almost never needed. Not even
+for special values like infinity (`Inf`) or not-a-number (`NaN`). Note
+that the letter case of these magic values doesn't matter so `inf` and
+`Inf` are equivalent.
 
-# List and Map
+Numeric literals can be specified using
+
+* decimal notation (e.g., `10`),
+* hexadecimal notation (e.g., `0xA`),
+* octal notation (e.g., `0o12`),
+* binary notation (e.g., `0b1010`),
+* floating point notation (e.g., `10.0`), and
+* scientific notation (e.g., `1.0e1`).
+
+All of those numeric literals are equivalent. However, note that all of
```

Comment:
You have already documented above that strings that can be parsed as numbers are automatically converted to numbers by builtins. Better not to repeat that.

In fact, since this is a convention of the standard builtins, not a property of the number type itself, it is better to

1. Consolidate all the explanations in builtin.md;
2. At the end of this section here, add a link to the builtin.md section as a "tip".

Reply:
I decided to just remove this paragraph. Eventually we'll want to document the behavior of `==` and `!=`. That is, whether or not they apply an appropriate epsilon, but that should be done in a separate change after we've decided whether we want the current behavior or for the commands to be friendlier for people who are naive about the pitfalls of comparing floating point quantities.